corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html

```diff
@@ -274,6 +274,28 @@ <h6>Example:</h6>
           </div>
         {% endif %}
       {% endif %}
+      {% if app.master_is_remote and not app.multimedia_pulled %}
+        <div class="panel panel-appmanager panel-appmanager-form">
+          <form action="{% url "pull_missing_multimedia" domain app.id %}" method="POST">
+            {% csrf_token %}
+            <legend>
+              {% trans "Pull missing multimedia" %}
+            </legend>
+            <div class="panel-body">
+              <div class="checkbox">
+                <label>
+                  <input type="checkbox" name="notify">
+                  {% trans "Email when finished (recommended for applications with large number of multimedia files or heavy files)" %}
```

Comment:
Did you consider implementing a progress bar instead?  If this takes something like 15 minutes, it'd be nice to have an indication of progress so you're not left wondering whether the task started or failed silently.

Reply:
it took like 2 hours last time and otherwise takes approx 1 hour.
That place is worth an improvement of its own, yes. @orangejenny  has suggested a better way to find out missing multimedia files.
But having a way to pull missing multimedia files sounds worth having irrespective of it being done within pull or after it.